examples/nmf/script.jl

```diff
@@ -0,0 +1,347 @@
+        alg,
+        options;
+        autodiff=:forward,
+    )
+    return getprimal_dict(Λ, Optim.minimizer(opt), ρ2)
+end;
+
+#
+# ## Example: noisy univariate Gaussians
+# 
+# We set up each observation as a mixture of 3 Gaussians with means sampled from `N(6, σ), N(0, σ), N(-6, σ)` respectively, and mixture weights sampled uniformly from `[0, 1]`. The resulting mixture model is normalised to sum to 1 on the discrete domain `coord`. 
+#
+f(x, μ, σ) = exp.(-(x .- μ) .^ 2)
+coord = range(-12, 12; length=100)
+N = 100
+σ = 1
+X = hcat(
+    [
+        rand() * f(coord, σ * randn() + 6, 1) +
+        rand() * f(coord, σ * randn(), 1) +
+        rand() * f(coord, σ * randn() - 6, 1) for _ in 1:N
+    ]...,
+)
+X = simplex_norm!(X);
+# 
+# We visualise the  observations. 
+#
+plot(coord, X; alpha=0.1, color=:blue, title="Input data", legend=nothing)
+# 
+# We can apply NMF with a squared Frobenius loss using the NMF.jl package. We seek `k = 3` components.  This performs poorly, since the pointwise nature of the loss function cannot handle the translational noise in the data.
+# 
+k = 3
+result = nnmf(X, k; alg=:multmse)
+plot(coord, result.W; title="NMF with Frobenius loss", palette=:Set1_3)
+#
+# We can now set up a cost matrix corresponding to the domain `coord`. 
+#
+C = pairwise(SqEuclidean(), coord)
+C = C / mean(C);
+# Specify parameters
+ε = 0.025
+ρ1, ρ2 = (5e-2, 5e-2);
+# Compute Gibbs kernel 
+K = exp.(-C / ε);
+# 
+# Now we use a random initialisation, where columns of `D` and `Λ` are normalised to sum to 1. 
+# 
+D = rand(size(X, 1), k) # dictionary 
+simplex_norm!(D; dims=1) # norm columnwise
+Λ = rand(k, size(X, 2)) # weights
+simplex_norm!(Λ; dims=1); # norm rowwise
+# 
+# We now run 10 iterations of Wasserstein-NMF.
+#
+n_iter = 10
+for iter in 1:n_iter
+    @info "Wasserstein-NMF: iteration $iter"
+    D .= solve_dict(
+        X,
+        K,
+        ε,
+        Λ,
+        ρ2;
+        alg=LBFGS(),
+        options=Optim.Options(;
+            iterations=250, g_tol=1e-4, show_trace=false, show_every=10
+        ),
+    )
+    Λ .= solve_weights(
+        X,
+        K,
+        ε,
+        D,
+        ρ1;
+        alg=LBFGS(),
+        options=Optim.Options(;
+            iterations=250, g_tol=1e-4, show_trace=false, show_every=10
+        ),
+    )
+end
+# 
+# We observe that Wasserstein-NMF learns atoms (columns of $D$) corresponding to the three Gaussians used to generate the input data.
+#
+plot(coord, D; title="NMF with Wasserstein loss", palette=:Set1_3)
+
+# ## Example: image data (MNIST)
+# Here we will download MNIST dataset using MLDatasets.jl and downscale each image to 14x14 to allow for faster runtime (since we are running on CPU). 
+#
+sizex, sizey = 28, 28
+factor = 2 # downscale factor 
+Σ = hcat([sum(I(sizex)[:, i:(i + factor - 1)]; dims=2) for i in 1:factor:sizex]...)
+sizex, sizey = sizex ÷ factor, sizey ÷ factor
+N = 256
+# must set this for Documenter
+ENV["DATADEPS_ALWAYS_ACCEPT"] = true
```

Comment:
We could just set it in the Github action.

Reply:
Done!